apps/browser/src/platform/services/popup-view-cache-background.service.ts

```diff
@@ -61,7 +61,20 @@ export class PopupViewCacheBackgroundService {
     merge(
       // on tab changed, excluding extension tabs
       fromChromeEvent(chrome.tabs.onActivated).pipe(
-        switchMap(([tabInfo]) => BrowserApi.getTab(tabInfo.tabId)),
+        switchMap(async (tabs) => {
+          const tab = await BrowserApi.getTab(tabs[0].tabId);
+
+          // FireFox sets the `url` to "about:blank" and won't populate the `url` until the `onUpdated` event
+          if (tab.url !== "about:blank") {
+            return tab;
+          }
+
+          return firstValueFrom(
+            fromChromeEvent(chrome.tabs.onUpdated).pipe(
+              switchMap(([tabId]) => BrowserApi.getTab(tabId)),
+            ),
+          );
```

Comment:
issue: don't convert the observable into a promise. The outer `switchMap` (line 64) will just turn it back into an observable again, but this time without proper unsubcription logic

```suggestion
          return fromChromeEvent(chrome.tabs.onUpdated).pipe(
              first()
              switchMap(([tabId]) => BrowserApi.getTab(tabId)),
            );
```